### PR TITLE
Enable constrained gridded reading for observations in Colocator

### PR DESCRIPTION
**Pull request by jgliss**
This is just a quick fix to make a first version work. In principle, this should also be possible to be applied to the model data, both in gridded/gridded and gridded/ungridded colocation. Can thus, only be  considered a limited fix of #152.
Includes also some cleanup for #154.

### CODE REVIEW

File: pyaerocom/colocation_auto.py [hunk 1]
```diff
@@ -948,7 +909,7 @@ def _run_gridded_gridded(self, var_name=None):
         data_objs = {}
 
         for model_var, obs_var in var_matches.items():
-
+            obs_filters_var = obs_filters[obs_var] if obs_var in obs_filters else {}
             print_log.info('Running {} / {} ({}, {})'.format(self.model_id,
```
Comment (ejgal) [inline pyaerocom/colocation_auto.py:913]: `obs_filters` might not be defined here.

File: pyaerocom/colocation_auto.py [hunk 2]
```diff
@@ -920,22 +893,10 @@ def _run_gridded_gridded(self, var_name=None):
         obs_reader = ReadGridded(self.obs_id)
 
         if 'obs_filters' in self:
-            remaining_filters = self._eval_obs_filters()
-            if bool(remaining_filters):
-                raise NotImplementedError('Cannot apply filters {} to gridded '
-                                          'observation data.'.format(remaining_filters))
+            obs_filters = self._eval_obs_filters()
```
Comment (ejgal) [inline pyaerocom/colocation_auto.py:896]: Could do: `obs_filters = self._eval_obs_filters() if 'obs_filters' in self else {}` to ensure `obs_filters` is set.

Comment (jgliss) [reply pyaerocom/colocation_auto.py:832]: Done now, had some conflicts in colocation_auto.py after merging v090dev, and had to fix manually. Hope everything still works.

**Comment by jgliss** [comment]
@ejgal can you check once more?

**Pull request merged**